network/kademlia.go

```diff
@@ -1061,6 +1069,19 @@ func (k *Kademlia) connectedNeighbours(peers [][]byte) (got bool, n int, missing
 	return gots == len(peers), gots, culprits
 }
 
+//calculates the expected min size of a given bin (minBinSize)
+//binSize = k.MinBinSize * 2 ^ (depth - binNumber - 1)
+func (k *Kademlia) expectedMinBinSize(binNumber int) int {
```

Comment:
I think we need to add here a maximum bin size. If the deep of the kademlia is too deep, the 0 bin will have too many peers.